### PR TITLE
Draft: Improve types

### PR DESCRIPTION
**Pull request by raycharius**
###  Summary

Draft pull request to discussion in [this issue](https://github.com/slackapi/node-slack-sdk/issues/1227) around improving @slack/types. See discussion [here](https://github.com/slackapi/node-slack-sdk/issues/1227).

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/types/src/index.ts [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
+  Channels = 'channels',
+  Conversations = 'conversations',
+  External = 'external',
+}
+
 export interface Dialog {
   title: string;
   callback_id: string;
   elements: {
-    type: 'text' | 'textarea' | 'select';
```
Comment (seratch) [inline packages/types/src/index.ts:8]: I'm afraid that changing union type to enum may be fully backward-compatible. I cannot tell what the possible troubles are by this but at least, unions and enums are not the same. To me, just naming the union types would be a safer refactoring. What do you think?

Comment (rr-codes) [reply packages/types/src/index.ts:8]: Just my two cents, but I really love enums more than unions aha

**Comment by raycharius** [comment]
I've been playing around with this a bit, and you are totally right. For some reason, I always though of enums as a subset of union types, but it will not work with simple strings. 

Naming the union types would definitely make the code more structured, but that would be a change that's more internal for your team as opposed to offering value for developers using the SDK. 

There is one other possibility, to have those values available as imports – export an object literal instead of an enum and create union types that reference the values in those objects. Advantage would be having those available as constants instead of working with strings for those using the package, but not sure if that's in scope.

**Comment by raycharius** [comment]
What are thoughts on the discriminating union types for more strict type checking? 

Advantages: 

- A much better and more type-safe experience with the package and SDK, such as with the `WebClient`.

Disadvantages: 

- Would require more swift updates to the package if API validation changes.
- It could be a breaking change, depending on how breaking is defined. It could throw compiler errors for objects that don't comply with the API validation, but unless there are some cases that I'm not taking into consideration, it would readonly only be showing issues with current code that would probably be rejected by the API anyway.

**Comment by filmaj** [comment]
I am closing this pull request - but have referenced the commits in it extensively! - in favour of having the different discussions about each type of change in this PR separately. They are:

- https://github.com/slackapi/node-slack-sdk/issues/1816
- https://github.com/slackapi/node-slack-sdk/issues/1817
- https://github.com/slackapi/node-slack-sdk/issues/1818

Thank you very much for your work in this PR. I know it has dragged on for literal years, but I think we will actually incorporate at least some of the suggestions from this work into the project!

**Pull request closed without merging**